sunpy/net/dataretriever/client.py

```diff
@@ -225,7 +176,63 @@ def _can_handle_query(cls, *query):
         `sunpy.net.fido_factory.UnifiedDownloaderFactory`
         class uses to dispatch queries to this Client.
         """
-        raise NotImplementedError
+        from sunpy.net import attrs as a
+
+        required = {a.Time, a.Instrument}
+        adict = cls.register_values()
+        optional = {i for i in adict.keys()} - required
+        if hasattr(cls, 'required'):
+            required = cls.required
+        if hasattr(cls, 'optional'):
+            optional = cls.optional
+        if not cls.check_attr_types_in_query(query, required, optional):
+            return False
+        for key in adict:
+            all_vals = [i[0].lower() for i in adict[key]]
+            for x in query:
+                if isinstance(x, key) and str(x.value).lower() not in all_vals:
+                    return False
+        return True
+
+    def post_search_hook(self, exdict, matchdict):
+        """
+        Helper function used after `search()` which makes the
+        extracted metadata representable in a query response table.
+
+        Parameters
+        ----------
+        exdict: `dict`
+            Represents metadata extracted from files.
+        matchdict: `dict`
+            Contains attr values accessed from `register_values()`
+            and the search query itself.
+
+        Returns
+        -------
+        map_: `~collections.OrderedDict`
+            An Ordered Dictionary which is used by `QueryResponse`
+            to show results.
+        """
+        map_ = OrderedDict()
+        almost_day = TimeDelta(1 * u.day - 1 * u.millisecond)
+        if 'month' in exdict and 'day' in exdict:
+            start = parse_time("{}/{}/{}".format(exdict['year'], exdict['month'], exdict['day']))
+            end = start + almost_day
+        elif 'year' in exdict:
+            start = parse_time("{}/{}/{}".format(exdict['year'], 1, 1))
+            end = parse_time("{}/{}/{}".format(exdict['year'], 12, 31)) + almost_day
+        map_['Time'] = TimeRange(start, end)
+        map_['Start Time'] = start.strftime(TIME_FORMAT)
+        map_['End Time'] = end.strftime(TIME_FORMAT)
```

Comment:
Why are we injecting the timerange and the start and end times, why not just one or the other?

Reply:
This was done so that we have `Time` as a Quantity in the table which can be used for inspection. And start time/endtime as strings which can be used for showing in the response table, so that results look similar as it was before for time.

If we inject only `Time`, we had to do that thing in `build_table`, and as I also had to preserve order of colnames in the table, it was relatively difficult to implement.

Reply:
Because scraper needs a `TimeRange` for generating urls.

Reply:
Also we would lose those milliseconds in endtime, if we use stringified time.